sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/implementation/credential/resolver/AzureTokenCredentialResolver.java

```diff
@@ -3,67 +3,84 @@
 
 package com.azure.spring.core.implementation.credential.resolver;
 
-import com.azure.identity.ClientCertificateCredential;
+import com.azure.core.credential.TokenCredential;
 import com.azure.identity.ClientCertificateCredentialBuilder;
-import com.azure.identity.ClientSecretCredential;
 import com.azure.identity.ClientSecretCredentialBuilder;
-import com.azure.identity.ManagedIdentityCredential;
 import com.azure.identity.ManagedIdentityCredentialBuilder;
 import com.azure.spring.core.aware.authentication.TokenCredentialAware;
 import com.azure.spring.core.credential.AzureCredentialResolver;
 import com.azure.spring.core.credential.provider.AzureTokenCredentialProvider;
 import com.azure.spring.core.properties.AzureProperties;
 import org.springframework.util.StringUtils;
 
+import java.util.function.Function;
+
 /**
  * Resolve the token credential according to the azure properties.
  */
-public final class AzureTokenCredentialResolver implements AzureCredentialResolver<AzureTokenCredentialProvider> {
+public class AzureTokenCredentialResolver implements AzureCredentialResolver<AzureTokenCredentialProvider> {
+
+    private final Function<AzureProperties, TokenCredential> resolveFunction;
+
+    public AzureTokenCredentialResolver(Function<AzureProperties, TokenCredential> resolveFunction) {
+        this.resolveFunction = resolveFunction;
+    }
+
+    public AzureTokenCredentialResolver() {
+        this.resolveFunction = AzureTokenCredentialResolver::resolveTokenCredential;
+    }
 
     @Override
     public AzureTokenCredentialProvider resolve(AzureProperties properties) {
-        // TODO (xiada): the token credential logic
-        final TokenCredentialAware.TokenCredential credential = properties.getCredential();
-        final String tenantId = properties.getProfile().getTenantId();
-        if (credential == null) {
+        TokenCredential tokenCredential = this.resolveFunction.apply(properties);
+        return tokenCredential == null ? null : new AzureTokenCredentialProvider(tokenCredential);
```

Comment:
What is the difference of responsibility between xxxProvider and xxxBuilder? and why this class name is AzureTokenCredentialResolver while this method return type of AzureTokenCredentialProvider?

Reply:
I think this is a good point, I can try to get rid of the credential provider.